tests/Functional/WriteTest.php

```diff
@@ -307,4 +323,30 @@ public function testDeleteWhereIsNull(): void
 
         self::assertCount(0, $data);
     }
+
+    private function expectPreviousException(string $expectedExceptionClass, Closure $test): void
+    {
+        try {
+            $test();
+        } catch (Throwable $exception) {
+            $e = $exception;
+            do {
+                if ($e instanceof $expectedExceptionClass) {
+                    $this->addToAssertionCount(1);
+
+                    return;
+                }
+
+                $e = $e->getPrevious();
+            } while ($e);
```

Comment:
This way, the test allows the driver exception to be at any nesting level which is kinda odd: we know exactly where it's supposed to be. Another idea is: since we're dealing with the logic exceptions here (identity columns not supported or one hasn't been initialized in the current session), we may just test that a logic exception is thrown.